### ISSUE TITLE
Apparent error in wavelength subsetting 

### ISSUE THREAD

**Issue by serbinsh** [meireles/spectrolab#8]
Error message:

```
Error in i_match_label(wavelengths(x), j) : 
  Following labels not found: 40040140240340440540640740841041141241341441541641741841942042142242342442642742943043143243343443543643743843944044244344444544644744844945045145245345545645745845946046146246346446546646846947047147247347447547647747847948048148248348448548648748848949149249349449549649749849950050150250350450550650850951051151251351451551651851952052152252352552652752852953053153253353453553653753853954054154254354454554654754854955055155255355455555655755855956056156256356456556656756856957057157257457557657757857958158258358458558658758858959059259359459559659759859960060160360460560660760860961061161261461561661761861962062162262362462562662762963063163263363463563663763863964064164264364464564664864965065165265365465565665765865966066166266366466566666766866967067167267367467567667767867968068168268368468568668768868969069169269369469569669769869970070170270370470570670770870971071171271371471571671771871972072172272472572672872973073173273373473573673773873
```

function ran:

```
spec_sub_vnir <- spec.files[ , 400:1200 ]
```

wavelengths:

```
> wavelengths(spec.files)
   [1]  339.0  340.4  341.9  343.3  344.8  346.2  347.7  349.1  350.6  352.1  353.5  355.0  356.4  357.9  359.4  360.8  362.3  363.7
  [19]  365.2  366.7  368.1  369.6  371.0  372.5  374.0  375.4  376.9  378.4  379.8  381.3  382.7  384.2  385.7  387.1  388.6  390.0
  [37]  391.5  393.0  394.4  395.9  397.3  398.8  400.3  401.7  403.2  404.6  406.1  407.6  409.0  410.5  411.9  413.4  414.9  416.3
  [55]  417.8  419.2  420.7  422.1  423.6  425.0  426.5  428.0  429.4  430.9  432.3  433.8  435.2  436.7  438.1  439.6  441.0  442.5
  [73]  443.9  445.4  446.8  448.3  449.7  451.1  452.6  454.0  455.5  456.9  458.4  459.8  461.2  462.7  464.1  465.6  467.0  468.5
  [91]  469.9  471.3  472.8  474.2  475.6  477.1  478.5  479.9  481.4  482.8  484.2  485.7  487.1  488.5  490.0  491.4  492.8  494.2
 [109]  495.7  497.1  498.5  499.9  501.4  502.8  504.2  505.6  507.0  508.5  509.9  511.3  512.7  514.1  515.6  517.0  518.4  519.8
 [127]  521.2  522.6  524.0  525.5  526.9  528.3  529.7  531.1  532.5  533.9  535.3  536.7  538.1  539.5  540.9  542.3  543.7  545.1
 [145]  546.5  547.9  549.3  550.7  552.1  553.5  554.9  556.3  557.7  559.1  560.5  561.9  563.3  564.7  566.1  567.5  568.9  570.3
 [163]  571.7  573.0  574.4  575.8  577.2  578.6  580.0  581.3  582.7  584.1  585.5  586.9  588.3  589.6  591.0  592.4  593.8  595.2
 [181]  596.5  597.9  599.3  600.7  602.0  603.4  604.8  606.1  607.5  608.9  610.3  611.6  613.0  614.4  615.7  617.1  618.5  619.8
 [199]  621.2  622.6  623.9  625.3  626.7  628.0  629.4  630.7  632.1  633.5  634.8  636.2  637.5  638.9  640.2  641.6  642.9  644.3
 [217]  645.7  647.0  648.3  649.7  651.1  652.4  653.8  655.1  656.5  657.8  659.2  660.5  661.8  663.2  664.5  665.9  667.2  668.6
 [235]  669.9  671.2  672.6  673.9  675.3  676.6  677.9  679.3  680.6  681.9  683.3  684.6  685.9  687.3  688.6  689.9  691.3  692.6
 [253]  693.9  695.3  696.6  697.9  699.2  700.6  701.9  703.2  704.5  705.9  707.2  708.5  709.8  711.2  712.5  713.8  715.1  716.4
 [271]  717.8  719.1  720.4  721.7  723.0  724.3  725.6  727.0  728.3  729.6  730.9  732.2  733.5  734.8  736.1  737.4  738.7  740.0
 [289]  741.3  742.6  743.9  745.2  746.5  747.8  749.1  750.4  751.7  753.0  754.3  755.6  756.9  758.2  759.5  760.8  762.1  763.4
 [307]  764.7  766.0  767.2  768.5  769.8  771.1  772.4  773.7  775.0  776.2  777.5  778.8  780.1  781.3  782.6  783.9  785.2  786.5
 [325]  787.7  789.0  790.3  791.5  792.8  794.1  795.4  796.6  797.9  799.2  800.4  801.7  803.0  804.2  805.5  806.7  808.0  809.2
 [343]  810.5  811.8  813.0  814.3  815.5  816.8  818.0  819.3  820.5  821.8  823.0  824.3  825.5  826.8  828.0  829.2  830.5  831.7
 [361]  833.0  834.2  835.4  836.7  837.9  839.2  840.4  841.6  842.8  844.1  845.3  846.5  847.8  849.0  850.2  851.4  852.7  853.9
 [379]  855.1  856.3  857.5  858.8  860.0  861.2  862.4  863.6  864.8  866.0  867.2  868.5  869.7  870.9  872.1  873.3  874.5  875.7
 [397]  876.9  878.1  879.3  880.5  881.7  882.9  884.1  885.3  886.5  887.7  888.8  890.0  891.2  892.4  893.6  894.8  896.0  897.1
 [415]  898.3  899.5  900.7  901.9  903.0  904.2  905.4  906.6  907.8  908.9  910.1  911.3  912.4  913.6  914.8  915.9  917.1  918.3
 [433]  919.4  920.6  921.8  922.9  924.1  925.2  926.4  927.6  928.7  929.9  931.0  932.2  933.3  934.5  935.7  936.8  938.0  939.1
 [451]  940.3  941.4  942.5  943.7  944.8  946.0  947.1  948.3  949.4  950.6  951.7  952.8  954.0  955.1  956.3  957.4  958.6  959.7
 [469]  960.8  962.0  963.1  964.2  965.4  966.5  967.7  968.8  969.9  971.1  972.2  973.3  974.5  975.6  976.7  977.9  979.0  980.1
 [487]  981.3  982.4  983.5  984.7  985.8  986.9  988.1  989.2  990.3  991.5  992.6  993.7  994.9  996.0  997.1  998.3  999.4 1000.5
 [505] 1001.7 1002.8 1003.9 1005.1 1006.2 1007.4 1008.5 1009.6  972.1  975.9  979.8  983.6  987.5  991.3  995.1  999.0 1002.8 1006.6
 [523] 1010.5 1014.3 1018.1 1022.0 1025.8 1029.6 1033.4 1037.3 1041.1 1044.9 1048.7 1052.5 1056.4 1060.2 1064.0 1067.8 1071.6 1075.4
 [541] 1079.2 1083.1 1086.9 1090.7 1094.5 1098.3 1102.1 1105.9 1109.7 1113.5 1117.3 1121.1 1124.9 1128.7 1132.5 1136.3 1140.1 1143.9
 [559] 1147.7 1151.4 1155.2 1159.0 1162.8 1166.6 1170.4 1174.2 1177.9 1181.7 1185.5 1189.3 1193.1 1196.8 1200.6 1204.4 1208.2 1211.9
 [577] 1215.7 1219.5 1223.2 1227.0 1230.8 1234.5 1238.3 1242.0 1245.8 1249.6 1253.3 1257.1 1260.8 1264.6 1268.3 1272.1 1275.8 1279.6
 [595] 1283.3 1287.1 1290.8 1294.6 1298.3 1302.1 1305.8 1309.5 1313.3 1317.0 1320.7 1324.5 1328.2 1331.9 1335.7 1339.4 1343.1 1346.9
 [613] 1350.6 1354.3 1358.0 1361.8 1365.5 1369.2 1372.9 1376.6 1380.3 1384.1 1387.8 1391.5 1395.2 1398.9 1402.6 1406.3 1410.0 1413.7
 [631] 1417.4 1421.1 1424.8 1428.5 1432.2 1435.9 1439.6 1443.3 1447.0 1450.7 1454.4 1458.1 1461.8 1465.5 1469.2 1472.8 1476.5 1480.2
 [649] 1483.9 1487.6 1491.2 1494.9 1498.6 1502.3 1505.9 1509.6 1513.3 1517.0 1520.6 1524.3 1528.0 1531.6 1535.3 1539.0 1542.6 1546.3
 [667] 1549.9 1553.6 1557.2 1560.9 1564.6 1568.2 1571.9 1575.5 1579.2 1582.8 1586.4 1590.1 1593.7 1597.4 1601.0 1604.7 1608.3 1611.9
 [685] 1615.6 1619.2 1622.8 1626.5 1630.1 1633.7 1637.3 1641.0 1644.6 1648.2 1651.8 1655.5 1659.1 1662.7 1666.3 1669.9 1673.6 1677.2
 [703] 1680.8 1684.4 1688.0 1691.6 1695.2 1698.8 1702.4 1706.0 1709.6 1713.2 1716.8 1720.4 1724.0 1727.6 1731.2 1734.8 1738.4 1742.0
 [721] 1745.6 1749.2 1752.8 1756.3 1759.9 1763.5 1767.1 1770.7 1774.2 1777.8 1781.4 1785.0 1788.5 1792.1 1795.7 1799.2 1802.8 1806.4
 [739] 1809.9 1813.5 1817.1 1820.6 1824.2 1827.7 1831.3 1834.8 1838.4 1841.9 1845.4 1849.0 1852.5 1856.0 1859.6 1863.1 1866.6 1870.1
 [757] 1873.6 1877.1 1880.6 1884.1 1887.6 1891.1 1894.6 1898.1 1901.6 1905.1 1908.5 1912.0 1898.8 1901.6 1904.3 1907.0 1909.7 1912.5
 [775] 1915.2 1917.9 1920.6 1923.3 1926.0 1928.7 1931.4 1934.1 1936.8 1939.5 1942.2 1944.9 1947.6 1950.3 1952.9 1955.6 1958.3 1961.0
 [793] 1963.6 1966.3 1969.0 1971.6 1974.3 1977.0 1979.6 1982.3 1984.9 1987.6 1990.2 1992.8 1995.5 1998.1 2000.8 2003.4 2006.0 2008.6
 [811] 2011.3 2013.9 2016.5 2019.1 2021.7 2024.3 2027.0 2029.6 2032.2 2034.8 2037.4 2040.0 2042.6 2045.2 2047.7 2050.3 2052.9 2055.5
 [829] 2058.1 2060.6 2063.2 2065.8 2068.4 2070.9 2073.5 2076.0 2078.6 2081.1 2083.7 2086.2 2088.8 2091.3 2093.9 2096.4 2098.9 2101.5
 [847] 2104.0 2106.5 2109.0 2111.6 2114.1 2116.6 2119.1 2121.6 2124.1 2126.6 2129.1 2131.6 2134.1 2136.6 2139.1 2141.5 2144.0 2146.5
 [865] 2149.0 2151.4 2153.9 2156.4 2158.8 2161.3 2163.8 2166.2 2168.7 2171.1 2173.6 2176.0 2178.4 2180.9 2183.3 2185.7 2188.2 2190.6
 [883] 2193.0 2195.4 2197.9 2200.3 2202.7 2205.1 2207.5 2209.9 2212.3 2214.7 2217.1 2219.5 2221.9 2224.2 2226.6 2229.0 2231.4 2233.8
 [901] 2236.1 2238.5 2240.9 2243.2 2245.6 2247.9 2250.3 2252.6 2255.0 2257.3 2259.7 2262.0 2264.4 2266.7 2269.0 2271.4 2273.7 2276.0
 [919] 2278.3 2280.6 2282.9 2285.2 2287.6 2289.9 2292.2 2294.5 2296.8 2299.1 2301.3 2303.6 2305.9 2308.2 2310.5 2312.8 2315.0 2317.3
 [937] 2319.6 2321.8 2324.1 2326.3 2328.6 2330.9 2333.1 2335.4 2337.6 2339.8 2342.1 2344.3 2346.5 2348.8 2351.0 2353.2 2355.4 2357.7
 [955] 2359.9 2362.1 2364.3 2366.5 2368.7 2370.9 2373.1 2375.3 2377.5 2379.7 2381.9 2384.0 2386.2 2388.4 2390.6 2392.8 2394.9 2397.1
 [973] 2399.2 2401.4 2403.6 2405.7 2407.9 2410.0 2412.2 2414.3 2416.4 2418.6 2420.7 2422.9 2425.0 2427.1 2429.2 2431.4 2433.5 2435.6
 [991] 2437.7 2439.8 2441.9 2444.0 2446.1 2448.2 2450.3 2452.4 2454.5 2456.6
 [ reached getOption("max.print") -- omitted 24 entries ]```

**Comment by serbinsh** [comment]
Aha, I see....

```
spec_sub_vnir <- spec.files[ , wavelengths(spec.files, 400, 1200) ]
```

because I am using the raw channels from the SVC.  The vignette is a little confusing (at least to me) because the steps show reading in SVC data, having 1024 channels and then subsetting like about.  Buy you have to keep reading to find subsetting without 1nm wavelengths.  

You have to note that the spectra species changes within the examples.  Sooo this isnt really a bug, but perhaps something to think about WRT docs?

**Comment by meireles** [comment]
You're right that this is confusing. Providing better examples and a **sensible error message** just made our to-do list.

**Comment by serbinsh** [comment]
OK thanks.  Also sorry for some of the grammar in the thread, looks like I was typing quickly or had some auto-correct issues!

**Comment by meireles** [comment]
hopefully fixed with a7c5cc37db7244c5e7eb236ad751711caba6c3e5
